api/peripheral_api/models/RFID.js

```diff
@@ -0,0 +1,27 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const RFIDSchema = new Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+    },
+    byte: {
+      type: String,
+      unique: true,
+      required: true,
+    },
+    created: {
+      type: Date,
+      default: Date.now,
+    },
+    lastScanned : {
+      type: Date,
+      default: Date.now,
+    }
+  },
+  { collection: 'RFID' }
+);
+
```

Comment:
It would be best if we don't store the user's true card data, but instead a hash of it.

This can look something like:

```js
RfidSchema.pre('save', function(next) {
  if (this.isModified('byte') || this.isNew) {
    bcrypt.genSalt(10, function(error, salt) {
      if (error) {
        return next(error);
      }
      bcrypt.hash(this.byte, salt, function(error, hash) {
        if (error) {
          return next(error);
        }
        this.byte = hash;
      });
    });
  }
  return next();
});
```

also would need custom logic to compare a card with the hash. I stole the above example from [`api/main_endpoints/models/User.js`](https://github.com/SCE-Development/Core-v4/blob/371f90a139ad30132155c2ab7e0eaf2ac9926376/api/main_endpoints/models/User.js#L77). You should also grab the compare logic from there too.

Reply:
So when the user scans the card we will have to hash the incoming byte and then look for the hashed byte in the database

Reply:
yes sir, similar to how we do passwords in the above link

Reply:
heres an example of us comparing passwords when the user logs in: [`api/main_endpoints/routes/Auth.js#L83`](https://github.com/SCE-Development/Core-v4/blob/371f90a139ad30132155c2ab7e0eaf2ac9926376/api/main_endpoints/routes/Auth.js#L83)

Reply:
Storing the information w/ a hash sounds like it would be optimal for what we need. +1

Reply:
OKay I will implement it before making pr for the API handlers and pub-sub